### DISCUSSION TITLE
Utility typedefs and extensions

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#126] [Show and tell]
It could be tedious to work with unnamed Records so this could help:

```dart
typedef ValueManager<T> = (T, Setter<T>);

typedef ValueManagerF<T> = (Getter<T>, Setter<T>);

typedef Getter<T> = T Function();

typedef Setter<T> = void Function(T);

extension ValueManagerExtension<T> on ValueManager<T> {
  T get value => $1;

  T get() => $1;

  void set(T value) => $2(value);

  T setAndGet(T value) {
    $2(value);
    return value;
  }
}

extension ValueManagerFExtension<T> on ValueManagerF<T> {
  T get value => $1();

  T get() => $1();

  void set(T value) => $2(value);

  T setAndGet(T value) {
    $2(value);
    return value;
  }
}

```

**Comment by GregoryConrad** [comment]
Ohh, I think I would like something like:

```dart
extension ValueWrapper on GetterSetter {
 T get value => $1();
  void set value(T newValue) => $2(newValue);
}
```

Then all I'd have to do is add in a `use.data` side effect (that is identical to state getter/setter), and defer all the current state side effects to call that one. I'm not sure whether or not I'd want to deprecate the current suite of state/value side effects to remove redundancy, but that is definitely an option.

Thanks for bringing this up! I have a feeling I'll incorporate some of this when I tweak rebuilds-in-build.

**Reply by busslina** [reply]
Cool :)
